app/controllers/api/user_sessions_controller.rb

```diff
@@ -25,11 +7,14 @@ class Api::UserSessionsController < Api::BaseController
   respond_to :json
 
   def sync
-    data = JSON.parse(params[:data])
-    data = deep_symbolize(data)
+    form = Api::JsonForm.new(json: "{ \"data\": #{params[:data]} }", schema: Api::UserSessions::Schema, struct: Api::UserSessions::Struct)
```

Comment:
This is a bit hacky. If it's needed again let's think of an abstraction

Reply:
Any tips on how this abstraction might look? 🙏

Reply:
None. Let's think about that when the problem appears again